hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/cmd/FeedCommProcessor.java

```diff
@@ -185,17 +187,25 @@ public void run() {
                         String messageString = ApiDeserializer.toHawkularFormat(message);
                         Buffer buffer = new Buffer();
                         buffer.writeUtf8(messageString);
-                        FeedCommProcessor.this.webSocket.sendMessage(WebSocket.PayloadType.TEXT, buffer);
+                        RequestBody requestBody = RequestBody.create(WebSocket.TEXT, buffer.readByteArray());
+                        FeedCommProcessor.this.webSocket.sendMessage(requestBody);
                     } else {
                         BinaryData messageData = ApiDeserializer.toHawkularFormat(message,
                                 messageWithData.getBinaryData());
-                        BufferedSink sink = FeedCommProcessor.this.webSocket
-                                .newMessageSink(WebSocket.PayloadType.BINARY);
-                        try {
-                            emitToSink(messageData, sink);
-                        } finally {
-                            sink.close();
-                        }
+
+                        RequestBody requestBody = new RequestBody() {
+                            @Override
+                            public MediaType contentType() {
+                                return WebSocket.BINARY;
+                            }
+
+                            @Override
+                            public void writeTo(BufferedSink bufferedSink) throws IOException {
+                                    emitToSink(messageData, bufferedSink);
+                            }
+                        };
+
+                        FeedCommProcessor.this.webSocket.sendMessage(requestBody);
```

Comment:
For 1) if we created the sink (like before) then we should definitely close it, but since we are not the ones creating the sink we its dangerous to be closing it since we don't know what they want to be doing with the sink afterwards. Eg in the case of multipart messages, they need the sink to remain open since they need to write to it multiple times (https://github.com/square/okhttp/blob/parent-2.7.5/okhttp/src/main/java/com/squareup/okhttp/MultipartBuilder.java#L277) 

For 2) this passes the checkstyle for me either way. I have updated it so that it is more properly formatted.